src/package-operations.ts

```diff
@@ -167,15 +179,19 @@ export async function updatePackage(
  */
 function getUpdatedManifest(
   currentManifest: Partial<PackageManifest>,
-  updateSpecification: UpdateSpecification,
+  updateSpecification: UpdateSpecification &
+    Partial<MonorepoUpdateSpecification>,
 ) {
   const { newVersion, synchronizeVersions } = updateSpecification;
   if (synchronizeVersions) {
     // If we're synchronizing the versions of our updated packages, we also
     // synchronize their versions whenever they appear as a dependency.
     return {
       ...currentManifest,
-      ...getUpdatedDependencyFields(currentManifest, updateSpecification),
+      ...getUpdatedDependencyFields(
+        currentManifest,
+        updateSpecification as MonorepoUpdateSpecification,
```

Comment:
Nit: We could implement a type guard function instead (e.g. `isMonorepoUpdateSpecification`) and add it to the condition, rather than using `as` here.